analyzers/rust/rust.go

```diff
@@ -0,0 +1,110 @@
+// Package rust provides analysers for Rust projects.
+//
+// A `BuildTarget` in Rust is the directory of the Rust project containing `Cargo.lock`.
+
+package rust
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/apex/log"
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/fossas/fossa-cli/buildtools/cargo"
+	"github.com/fossas/fossa-cli/errors"
+	"github.com/fossas/fossa-cli/files"
+	"github.com/fossas/fossa-cli/graph"
+	"github.com/fossas/fossa-cli/module"
+	"github.com/fossas/fossa-cli/pkg"
+)
+
+type Analyzer struct {
+	Module  module.Module
+	Options Options
+}
+
+type Options struct {
+	Strategy string `mapstructure:"strategy"`
+}
+
+func New(m module.Module) (*Analyzer, error) {
+	log.WithField("options", m.Options).Debug("constructing analyzer")
+
+	var options Options
+	err := mapstructure.Decode(m.Options, &options)
+	if err != nil {
+		return nil, err
+	}
+	log.WithField("options", options).Debug("parsed analyzer options")
+
+	return &Analyzer{
+		Module:  m,
+		Options: options,
+	}, nil
+}
+
+// Discover constructs modules in all directories with a `Cargo.lock`.
+func Discover(dir string, options map[string]interface{}) ([]module.Module, error) {
+	var modules []module.Module
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			log.WithError(err).WithField("path", path).Debug("error while walking for discovery")
+		}
+
+		if !info.IsDir() && info.Name() == "Cargo.lock" {
+			moduleName := filepath.Base(path)
```

Comment:
I don't think `filepath.Base` is the function you're looking for here. It always returns "Cargo.lock". Consider taking the actual name from [`Cargo.toml`](https://doc.rust-lang.org/cargo/reference/manifest.html#the-manifest-format) or just:
```
abs, err := filepath.Abs(info.Name())
// if err ....
moduleName := filepath.Base(filepath.Dir(abs))
```

Reply:
Yea you're right, it will always return `Cargo.lock`. I'm hesitant to parse `Cargo.toml` here as it introduces another point of complexity and failure but doesn't really provide any information to the user. I agree that `Cargo.lock` isn't clear though so I will likely use the absolute path.